layers/layers.go

```diff
@@ -33,9 +34,13 @@ type Layers struct {
 }
 
 // Layer creates a Layer with a specified name.
-func (l Layers) Layer(name string) Layer {
```

Comment:
We had a back-and-forth on this previously and decided against automatically creating a directory when a layer was required.  Since some layers would be _contributed_ conditionally (but you might want to get a reference to it eagerly) automatically creating the directory would be leaving unnecessary artifacts on the filesystem.

As a programming practice, ensuring that the parent directory of a file exists before writing the file is broadly followed (you'll see us [encoding it as policy](https://github.com/buildpack/libbuildpack/blob/master/internal/write_file.go#L29)).  Therefore, we don't want to add this functionality.